### PR TITLE
Feature/28297 make error message at  a constant or a read only property  more specific

### LINKED ISSUE #28297: Error message at `a constant or a read-only property` could be more specific
**TypeScript Version:** 3.2.0-dev.20181101

**Code**

```ts
const isMutable = false;
isMutable = true;
```

**Expected behavior:**

`Cannot assign to 'isMutable' because it is a constant.`

**Actual behavior:**

`Cannot assign to 'isMutable' because it is a constant or a read-only property.` It's obviously not a property, and the compiler should be able to tell this. Note this is more tricky with `namespace N { export const x = 0; }`, which is also a property, but I think calling it just "a constant" in that case is fine.

### PR DESCRIPTION
**Pull request by a-tarasyuk**
Fixes #28297

### CODE REVIEW

File: src/compiler/diagnosticMessages.json [hunk 1]
```diff
@@ -2124,6 +2124,10 @@
         "category": "Error",
         "code": 2587
     },
+    "Cannot assign to '{0}' because it is a constant": {
```
Comment (andy-ms) [inline src/compiler/diagnosticMessages.json]: * Other error messages end in a period so this should too.
* We should also make the "read-only property" error message more specific, no need for "constant or"

File: src/compiler/checker.ts [hunk 2]
```diff
@@ -15891,7 +15891,12 @@ namespace ts {
                     return errorType;
                 }
                 if (isReadonlySymbol(localOrExportSymbol)) {
-                    error(node, Diagnostics.Cannot_assign_to_0_because_it_is_a_constant_or_a_read_only_property, symbolToString(symbol));
+                    if (isConstVariable(localOrExportSymbol)) {
```
Comment (andy-ms) [inline src/compiler/checker.ts]: I think this could just check `localOrExportSymbol.flags & SymbolFlags.Variable`

**Comment by a-tarasyuk** [comment]
@andy-ms Thanks. I've added requested changes.

**Comment by andy-ms** [comment]
@DanielRosenwasser please review (github reviewers section not working right now)

**Comment by DanielRosenwasser** [comment]
Thank you!

**Pull request merged**